pkg/securelaunch/eventlog/eventlog.go

```diff
@@ -0,0 +1,83 @@
+package eventlog
+
+import (
+	"fmt"
+	"github.com/9elements/tpmtool/pkg/tpm"
+	"github.com/u-root/u-root/pkg/mount"
+	slaunch "github.com/u-root/u-root/pkg/securelaunch"
+	"log"
+	"strings"
+)
+
+type EventLog struct {
+	Type     string `json:"type"`
+	Location string `json:"location"`
+}
+
+const (
+	eventLogFile        = "/sys/kernel/security/slaunch/eventlog"
+	defaultEventLogFile = "eventlog.txt" //only used if user doesn't provide any
+)
+
+func parseEvtLog(evtLogFile string) ([]byte, error) {
+
+	tpm.DefaultTCPABinaryLog = evtLogFile
+	firmware := tpm.Txt
+	TPMSpecVersion := tpm.TPM20
+	tcpaLog, err := tpm.ParseLog(firmware, TPMSpecVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	var w strings.Builder
+	for _, pcr := range tcpaLog.PcrList {
+		fmt.Fprintf(&w, "%s\n", pcr)
+		fmt.Fprintf(&w, "\n")
+	}
+	return []byte(w.String()), nil
+}
+
+// tpmtool parsed data --> written to --> eventlogPath
```

Comment:
You're going to need to fix all these functions:
// Persist blargs teh EvenLog with a snarf and returns any error
or some such.